### PR TITLE
Flow code storage docs improvements

### LINKED ISSUE #14387: All store flow code examples should be consolidated into the same page
### First check

- [X] I added a descriptive title to this issue.
- [X] I used GitHub search to find a similar request and didn't find it 😇

### Describe the issue

We have a dedicated page showing how to store flow code. https://docs-3.prefect.io/3.0rc/deploy/infrastructure-concepts/store-flow-code

However, some of the other deploy docs have similar examples that need to be merged into this page.

### Describe the proposed change

The following sections need to be merged into the store flow code page:

* https://docs-3.prefect.io/3.0rc/deploy/infrastructure-examples/docker#store-your-code-in-git-based-cloud-storage
* https://docs-3.prefect.io/3.0rc/deploy/infrastructure-examples/docker#store-your-code-in-cloud-provider-storage
* https://docs-3.prefect.io/3.0rc/deploy/run-flows-in-local-processes#retrieve-a-flow-from-remote-storage

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by discdiver**
Creates  Python `deploy` and `prefect.yaml` examples and reorganizes the page for more clear and comprehensive examples of flow code storage.

Advances work on #14387

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

**Review by zzstoatzz — CHANGES_REQUESTED** [review]
I think these examples need to be consolidated and fixed

- we should define a variable for `source` for use in `from_source` so its easier to read
- I think we can make use of panes/tabs to make better use of space

File: docs/3.0/deploy/infrastructure-concepts/store-flow-code.mdx [hunk 1]
```diff
@@ -152,164 +326,454 @@ This is intentional to avoid confusion about the git history and push process.
...
+        sourcce 
+        flow.from_source(
+            source="s3://my-bucket/my-folder",
+            entrypoint="my_file.py:my_flow",
+            credentials=aws_credentials
+        ).deploy(
+            name="my-aws-s3-deployment",
+            work_pool_name="my-work-pool"
+        )
+    ```
```
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/3.0/deploy/infrastructure-concepts/store-flow-code.mdx:487]: yeah. I'm still making changes here. I'll re-request review when all made.

File: docs/integrations/prefect-github/index.mdx [hunk 2]
```diff
@@ -65,16 +66,15 @@ from prefect_github import GitHubCredentials
 
 
 if __name__ == "__main__":
-    flow.from_source(
-        source=GitRepository(
+
+    source = GitRepository(
```
Comment (zzstoatzz) [inline docs/integrations/prefect-github/index.mdx:70]:
```suggestion
    git_repo = GitRepository(
```

File: docs/integrations/prefect-github/index.mdx [hunk 3]
```diff
@@ -65,16 +66,15 @@ from prefect_github import GitHubCredentials
...
         url="https://github.com/org/private-repo.git",
         credentials=GitHubCredentials.load("my-github-credentials-block")
-    ),
-    entrypoint="my_file.py:my_flow",
-    ).deploy(
+    )
+
+    flow.from_source(source=source, entrypoint="my_file.py:my_flow").deploy(
```
Comment (zzstoatzz) [inline docs/integrations/prefect-github/index.mdx:75]:
```suggestion
    flow.from_source(source=git_repo, entrypoint="my_file.py:my_flow").deploy(
```

**Review by zzstoatzz — APPROVED** [review]
looks good to me!

i have 2 requests:
- there's some incorrect indenting that I've highlighted
- can we use descriptive names for the `source`? like `git_repo` for example, so it reads more intuitively

File: docs/integrations/prefect-github/index.mdx (hunk 2 above)
Comment (zzstoatzz) [reply docs/integrations/prefect-github/index.mdx:70]: one example of what I mean

Comment (discdiver) [reply docs/integrations/prefect-github/index.mdx:70]: Sure thing.

File: docs/3.0/deploy/infrastructure-concepts/store-flow-code.mdx [hunk 4]
```diff
@@ -6,139 +6,316 @@ description: Learn about where to store your flow code.
...
-        repository: https://gitlab.com/org/my-private-repo.git
-        access_token: "{{ prefect.blocks.secret.my-block-name }}"
-```
+        flow.from_source(
+            source=source,
+            entrypoint="gh_credentials_block.py:my_flow",
+            ).deploy(
+                name="private-github-deploy",
+            work_pool_name="my_pool",
+        )
```
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/3.0/deploy/infrastructure-concepts/store-flow-code.mdx:85]: Thought I got all those. Good catch.

**Pull request merged**